components/omega/doc/design/MeshClass.md

```diff
@@ -0,0 +1,73 @@
+(omega-design-mesh)=
+# Mesh 
+
+## 1 Overview
+
+The mesh class will contain the YAKL arrays which describe the mesh and are used in the computation of the tendency terms in the discrete governing equations.
+
+## 2 Requirements
+
+### 2.1 Requirement: OMEGA will use the previously estabilished MPAS Mesh Spec
+
+The OMEGA mesh information should be compatiable with the [MPAS Mesh Specification](https://mpas-dev.github.io/files/documents/MPAS-MeshSpec.pdf).
+
+### 2.2 Requirement: Functionality is needed to read the mesh on the host and transfer relevant data to the device for comutation
+
+Not all mesh information is required in computing the tendency terms on the device, e.g. lonCell, latCell, etc. 
+However, other arrays will need to allocated and copied to the device for use in tendency computation.
+One option is to template the mesh class so a mesh object can be created for both the host and device, with a method to perform the copy between the two object types.
+
+### 2.3 Requirement: Zero-based cell, edge, and vertex numbering
+
+Although the existing MPAS Mesh spec uses a one-based mesh numbering, zero-based mesh numbering is required to be compatiable with the zero-based indexing used for YAKL arrays.
+
+### 2.4 Desired: Ability to support multiple independent mesh objects
+
+This flexibility is required to support future implmentations of spatially split barotropic and baroclinic modes that are computed on different resolution 
+
+### 2.5 Desired: OMEGA can read in a reduced number of mesh variables and compute the remaning array information online.
+
+Many of the mesh variables are not independent, e.g.  areaCell, weightsOnEdge, etc., and can be computed from a reduced set of mesh variables.
+This functionality could be used to reduce mesh/restart file size for high resoultion meshes.  
+Builiding in this flexibility would allow for all mesh-related calculations to be availiable within the code base instead of spreading them amonst various utility programs in MPAS-Tools.
+The functions used to compute the dependent mesh information can also be used to verify mesh information provided in the mesh input stream.
```

Comment:
That's a good point about reciprocals etc. I agree that the xyz coords and connectivity information is all we should need (i.e. jigsaw output). It seems like a lot of the [mpas_mesh_converter](https://github.com/MPAS-Dev/MPAS-Tools/blob/master/mesh_tools/mesh_conversion_tools/mpas_mesh_converter.cpp) functions could be included here since that's what we use in the meshing workflow to create all the other dependent mesh variables anyway.

Reply:
I agree that post processing is an important consideration of reducing mesh file info as well.

Reply:
The main requirement I would suggest adding for post-processing would be a capability (and a standard practice) of writing out all mesh variables to an output file as part of a simulation.  Although we could generate such variables on the fly during post processing, it seems desirable to have this done once as part of a model run.  Otherwise, the different post-processing runs might construct these variables differently (e.g. on different numbers of cores or with different compilers) and this would potentially complicate reproducibility of the post-processing.

Reply:
This makes sense and I will include it in the requirements section.